### PR TITLE
feat(NcAppNavigationCaption): Add `heading-id` prop to allow setting the ID on the caption itself

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

Add a prop to set caption id to the caption, needed for linking list and caption.

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

**Review by Pytal — APPROVED** [review]
🚀

**Comment by susnux** [comment]
Alternatives:
- Add `caption-id` prop that is always set on the name component.
- Add `id` as prop and bind to name component, and add `actions-id` and bind to the actions component

**Comment by ShGKme** [comment]
> * Add `caption-id` prop that is always set on the name component.

I'd go this way:
- We only need to set ID on the header (at least I don't see other cases)
- Seems very non-obvious behavior what attributes are proxied either on header or on actions menu trigger

And what if we need ID even when there are actions?

File: src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue [hunk 1]
```diff
@@ -148,6 +167,14 @@ export default {
...
 		},
 
+		/**
+		 * If `isHeading` is set, this defines the heading level that should be used
+		 */
+		headingLevel: {
+			type: Number,
+			default: 2,
+		},
+
```
Comment (ShGKme) [inline src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue:177]: A new feature then?

Comment (susnux) [reply src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue:177]: Probably. I need to fix a place that looks like this:
```
h2
  h2 < this should be h3 but is not possible
  ul < this is a appnavigation caption
```

**Comment by susnux** [comment]
> I'd go this way [...]

I agree, but I would additionally change the props handling, because I think we have a problem with Vue3 here as `class` and `style` now are assigned to the actions instead of the wrapper.

Also I do not think we need additional attributes on the actions but should keep them on either the wrapper or the inner name.

File: src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue (hunk 1 above)
Comment (susnux) [reply src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue]: Splitted the PR.

**Review by ShGKme — APPROVED** [review]

File: src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue [hunk 2]
```diff
@@ -139,6 +160,15 @@ export default {
...
 		},
 
+		/**
+		 * `id` to set on the inner caption
+		 * Can be used for connecting the `NcActionCaption` with `NcActionList` using `aria-labelledby`.
+		 */
+		captionId: {
+			type: String,
+			default: null,
+		},
```
Comment (ShGKme) [inline src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue:170]: `headingId` for consistency with `isHeading` and `headingLevel`

Comment (susnux) [reply src/components/NcAppNavigationCaption/NcAppNavigationCaption.vue]: I agree

**Review by Antreesy — APPROVED** [review]

**Comment by susnux** [comment]
> /backport to next

Did it manually: https://github.com/nextcloud-libraries/nextcloud-vue/pull/5580

**Pull request merged**